pkg/controller/job/job_controller.go

```diff
@@ -910,3 +941,31 @@ func errorFromChannel(errCh <-chan error) error {
 	}
 	return nil
 }
+
+// ensureJobConditionStatus appends or updates an existing job condition of the
+// given type with the given status value. Note that this function will not
+// append to the conditions list if the new condition's status is false
+// (because going from nothing to false is meaningless); it can, however,
+// update the status condition to false. The function returns a bool to let the
+// caller know if the list was changed (either appended or updated).
+func ensureJobConditionStatus(list []batch.JobCondition, cType batch.JobConditionType, status v1.ConditionStatus) ([]batch.JobCondition, bool) {
+	isUpdated, matches := false, false
+	for i := range list {
+		if list[i].Type == cType {
+			matches = list[i].Status == status
+			if !matches {
+				list[i].Status = status
+				list[i].LastTransitionTime = metav1.Now()
+				// The status value now matches what we want, so update `matches` to
+				// true. Also, since we updated the list, let's set `isUpdated`.
+				matches, isUpdated = true, true
+			}
+			// At this point, `matches` is always true. `isUpdated` may be false.
+			break
+		}
+	}
+	if !matches && status != v1.ConditionFalse {
+		return append(list, newCondition(cType, status, "", "")), true
```

Comment:
shouldn't we have a reason and message?

Reply:
I can see how reason/message makes sense for job failures, so I couldn't think of anything useful for reason/message for a job getting suspended, but I've added something anyway, PTAL :)